cmd/snap-confine/seccomp-support-test.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (C) 2015 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "seccomp-support-ext.c"
+#include "seccomp-support.c"
+
+#include <glib.h>
+#include <glib/gstdio.h>
+
+static void make_seccomp_profile(struct sc_seccomp_file_header *hdr, int *fd,
+				 char **fname)
+{
+	*fd = g_file_open_tmp(NULL, fname, NULL);
+	g_assert_true(*fd > 0);
+	int written = write(*fd, hdr, sizeof(struct sc_seccomp_file_header));
+	g_assert_true(written == sizeof(struct sc_seccomp_file_header));
+}
+
+static void test_must_read_and_validate_header_from_file__happy(void)
+{
+	struct sc_seccomp_file_header hdr = {
+		.header[0] = 'S',
+		.header[1] = 'C',
+		.version = 1,
+	};
+	char SC_CLEANUP(sc_cleanup_string) * profile = NULL;
+	int SC_CLEANUP(sc_cleanup_close) fd = 0;
+	make_seccomp_profile(&hdr, &fd, &profile);
+
+	FILE *SC_CLEANUP(sc_cleanup_file) file =
+	    sc_must_read_and_validate_header_from_file(profile, &hdr);
+	g_assert_true(file != NULL);
+}
+
+static void must_read_and_validate_header_from_file_dies_with(struct sc_seccomp_file_header hdr, const char *err_msg)
+{
+	if (g_test_subprocess()) {
+		char SC_CLEANUP(sc_cleanup_string) * profile = NULL;
+		int SC_CLEANUP(sc_cleanup_close) fd = 0;
+		make_seccomp_profile(&hdr, &fd, &profile);
+
+		FILE *SC_CLEANUP(sc_cleanup_file) file =
+		    sc_must_read_and_validate_header_from_file(profile, &hdr);
+		// the function above is expected to call die()
+		g_assert_not_reached();
+		// reference "file" to keep the compiler from warning
+		// that "file" is unused
+		g_assert_null(file);
+	}
+
+	g_test_trap_subprocess(NULL, 0, 0);
+	g_test_trap_assert_failed();
+	g_test_trap_assert_stderr(err_msg);
+}
+
```

Comment:
we should also check the file is missing case

Reply:
Done in [69400f9022f9dd3a10ac120deebc40304d864a10](https://github.com/snapcore/snapd/pull/13443/commits/69400f9022f9dd3a10ac120deebc40304d864a10)

It does not use `must_read_and_validate_header_from_file_dies_with` as the function signature is incompatible with the missing file case.